clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -46,11 +46,12 @@ class ApiClient(apiUrl: String, accessKey: String, secretKey: String) {
     if (metaRangeResp.isError) {
       throw new RuntimeException(s"failed to get meta_range ${metaRangeID}: [${metaRangeResp.code}] ${metaRangeResp.body}")
     }
-    val location = metaRangeResp.header("Location") match {
-      case Some(l) => l
-      case None => throw new RuntimeException("missing Location header in response for meta_range %s".format(metaRangeID))
+    val metaRangeLocation = parse(metaRangeResp.body) \ "location" match {
+      case JString(metaRangeLocation) => metaRangeLocation
+      case _ => // TODO(ariels): Bad parse exception type
+        throw new RuntimeException(s"expected property location in ${metaRangeResp.body}")
```

Comment:
Good news, everybody!  I'm getting rid of this entire manual parsing mess in favour of using the generated OpenAPI client #1837 as we speak!